scripts/build_go_slim.sh

```diff
@@ -110,10 +110,20 @@ for f in ./coder-slim_*; do
 done
 
 if [[ $compress != 0 ]]; then
-	log "--- Compressing coder-slim binaries using zstd level $compress ($dest_dir/coder.tar.zst)"
 	pushd "$dest_dir"
-	tar cf coder.tar coder-*
+	log "--- Generating SHA256 for coder-slim binaries ($dest_dir/coder.sha256)"
+	shasum -b -a 256 coder-* | tee coder.sha256
```

Comment:
Since we're only interested in integrity checking and not really tamper resistance, could we use a faster algorithm here? CRC32 for example takes like 1/6th (0.25 seconds vs 1.5 seconds) the amount of time as SHA256 does for all of the slim binaries

Reply:
That's a fair point, ironically SHA256 (currently) takes longer that `zstd -d`. Since it seems we aren't IO bound, we could verify the binaries concurrently. I'd be in favor of at least keeping `SHA1` for improved collision resistance (0.8s non-concurrently on my machine).

PS. I've always found it impossible to find proper (cross-platform) tools for doing CRC32 😅.

Reply:
I'm not super concerned about the generation taking an extra second, but not convinced that the collision resistance is worth an extra 1.25s startup time for coder server. What would be the main benefit of using a collision-less algorithm here? Our release binaries can already be verified using the checksums on GitHub releases, so this checksum is just to check that the extraction didn't have any errors right?

There is an official hash/crc32 package in stdlib so we wouldn't have to worry about cross-platform tools for performing the validation at least, and we could also use the go package for creating the checksums file too if we want.

Reply:
SHA1 should only add ~0.5s extra, and that could be even less with concurrency (I'll do some benchmarks).

> Our release binaries can already be verified using the checksums on GitHub releases, so this checksum is just to check that the extraction didn't have any errors right?

The purpose isn't really to verify the extraction, it's to verify that the file is the same without needing to (re-)decompress the archive. I'm not very knowledgable about CRC32 collisions, but we'd want to avoid one build matching another (older or newer) build.